source/extensions/filters/http/cache/cache_policy.h

```diff
@@ -0,0 +1,68 @@
+#pragma once
+
+#include "envoy/http/header_map.h"
+#include "envoy/stream_info/filter_state.h"
+
+#include "source/extensions/filters/http/cache/cache_headers_utils.h"
+#include "source/extensions/filters/http/cache/http_cache.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Cache {
+
+struct CacheEntryUsability {
+  CacheEntryStatus status = CacheEntryStatus::Unusable;
+  Envoy::Seconds age = Envoy::Seconds::max();
+};
+
+class CachePolicyCallbacks {
+public:
+  virtual ~CachePolicyCallbacks() = default;
+
+  virtual const Envoy::StreamInfo::FilterStateSharedPtr& filterState() PURE;
+};
+
+// CachePolicy is an extension point for deployment specific caching behavior.
+class CachePolicy {
+public:
+  virtual ~CachePolicy() = default;
+
+  // createCacheKey calculates the lookup key for storing the entry in the cache.
+  virtual Key createCacheKey(const Envoy::Http::RequestHeaderMap& request_headers) PURE;
+
+  // requestCacheable modifies the cacheability of the response during
```

Comment:
Use doxygen style comments for this interface. Based on the interface I think you mean to say:
```
/**
 * @return whether the request is cacheable.
 * /
```
Also should the method be const?

Reply:
I'm a bit wary of const at the interface level, as I don't want to inadvertently restrict what a custom CachePolicy implementation can do. That said, I personally have no plans to introduce anything that modifies CachePolicy in `createCacheKey`, `requestCacheable`, `responseCacheable` or `computeCacheEntryUsability`, and could add `const` to all 4 of those.

What do you think? Should I add const to all of the above?